python/hsfs/core/tfdata_engine.py

```diff
@@ -0,0 +1,461 @@
+except ModuleNotFoundError:
+    pass
+
+
+class TFDataEngine:
+    SUPPORTED_TFDTYPES = [
+        tf.float16,
+        tf.float32,
+        tf.float64,
+        tf.int16,
+        tf.int64,
+        tf.int32,
+    ]
+    SPARK_TO_TFDTYPES_MAPPINGS = {
+        "string": tf.string,
+        "short": tf.int16,
+        "int": tf.int32,
+        "long": tf.int64,
+        "float": tf.float32,
+        "double": tf.float64,
+    }
+
+    def __init__(
+        self,
+        training_dataset,
+        split,
+        target_name,
+        feature_names,
+        is_training,
+        cycle_length,
+    ):
+
+        self._training_dataset = training_dataset
+        self._split = split
+        self._target_name = target_name
+        self._feature_names = feature_names
+        self._is_training = is_training
+        self._cycle_length = cycle_length
+
+        self._features = self._training_dataset._features
+        self._training_dataset_format = self._training_dataset.data_format
+
+        self._input_files = _get_training_dataset_files(
+            self._training_dataset.location,
+            self._split,
+            filter_empty=True if self._training_dataset_format == "csv" else False,
+        )
+
+        if self._feature_names is None:
+            self._feature_names = [feat.name for feat in self._features]
+
+            if target_name in self._feature_names:
+                self._feature_names.remove(target_name)
+
+    def get_serialized_example_schema(self):
+        """
+        returns schema for parsing serialized example of ff record file
+        :return: schema for tfrecord file
+        :rtype: dict
+        """
+        _, tfrecord_feature_description = _get_tfdataset(
+            self._input_files, self._cycle_length
+        )
+        return tfrecord_feature_description
+
+    def tf_record_dataset(
+        self,
+        batch_size=None,
+        num_epochs=None,
+        one_hot_encode_labels=False,
+        num_classes=None,
+        process=False,
+        serialized_ndarray_fname=[],
+    ):
+        """
+        Reads tfrecord files and returns ParallelMapDataset or PrefetchDataset object ready to input model training.
+        Reads tfrecord files and returns ParallelMapDataset object if process argument set to False. ParallelMapDataset
+        object can be further processed before feeding into model training. If process argument set to True
+        tf_record_dataset will return PrefetchDataset object ready to input model training.
+        !!! example "Using tf_record_dataset"
+        Example of using tf_record_dataset:
+        ```python
+        connection = hsfs.connection()
+        fs = connection.get_feature_store();
+        td = fs.get_training_dataset("sample_model", 3)
+        td.tf_data(target_name = "id").tf_record_dataset(batch_size=1, num_epochs=1, process=True)
+        ```
+
+        # Arguments
+            batch_size: `Optional[int]`: Size of batch, defaults to `None`.
```

Comment:
```suggestion
            batch_size: Size of batch, defaults to `None`.
```

Types will be inserted automatically if you add them in the function signature as python types.